esphome/core/helpers.cpp

```diff
@@ -105,7 +105,7 @@ std::string truncate_string(const std::string &s, size_t length) {
 }
 
 std::string value_accuracy_to_string(float value, int8_t accuracy_decimals) {
-  auto multiplier = float(pow10(accuracy_decimals));
+  auto multiplier = float(pow(10.0f, accuracy_decimals));
```

Comment:
```suggestion
  auto multiplier = float(powf(10.0f, accuracy_decimals));
```

Whoops, don't know why the double version was used here. But we should fix it anyway.